check_commits.sh

```diff
@@ -105,18 +122,6 @@ function check_commit_for_signoffs_and_changelogs() {
         fi
     fi
 
-    if [ -n "${CHECK_CHANGELOGS}" ]; then
-        # Check that Changelog tags are present.
-        if ! echo "$COMMIT_MSG" | grep -i "^ *Changelog:" >/dev/null; then
-            echo >&2 "Commit ${i} doesn't have a changelog tag! Make a changelog entry for your commit (https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags)."
-            notvalid="$notvalid $i"
-        # Less than three words probably means something was misspelled, except for
-        # None, Title, Commit and All.
-        elif ! echo "$COMMIT_MSG" | egrep -i "^ *Changelog: *(None|Title|Commit|All|\S+(\s+\S+){2,}) *$" >/dev/null; then
-            echo >&2 "Commit ${i} has less than three words in its changelog tag! Typo? (https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags)."
```

Comment:
I will, note that the fix above does not already do what you say however :p

I guess we will have to do string distance for this. But it's fine. I'll write it :)

Reply:
https://github.com/mendersoftware/mendertesting/pull/179/commits/70bd631b1fcc068187924fe56b02d12bcdeb24d4

Reply:
> I will, note that the fix above does not already do what you say however :p

It doesn't...?

> I guess we will have to do string distance for this. But it's fine. I'll write it :)

I don't think this is a great fix, because it is both more complicated, and more error prone. With this fix it is not possible to write for example:

```
Changelog: Non valid keys are now detected.
```

because it will trigger on the word "non". And it would not trigger on something like `Changelog: TODO`.

Isn't the existing approach both easier and less error prone? If the string has only one or two words, and it isn't one of the allowed valued, it's an error.

Reply:
> It doesn't...?

Nope, how could it?

> Isn't the existing approach both easier and less error prone? If the string has only one or two words, and it isn't one of the allowed valued, it's an error.

Sure, but you commented that you wanted it to catch things like misspelled `Tilte`, etc.... This is the only way in that instance

Reply:
> Sure, but you commented that you wanted it to catch things like misspelled Tilte, etc.... This is the only way in that instance

I think we need face time for this, there is some misunderstanding here I think...

Reply:
Resolved f2f